io/disk/rawread.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM.
+# Author: Naresh Bannoth <nbannoth@in.ibm.com>
+
+"""
+Rawread test
+"""
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Rawread(Test):
+
+    """
+    Rawread is a benchmark suite that is aimed at performing a number
+    of simple tests of hard drive like write and read
+    """
+
+    def setUp(self):
+        """
+        checking install of required packages and extract and
+        compile of rawread suit.
+        """
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'libaio-devel']
+        if distro.detect().name == 'Ubuntu':
+            deps.extend(['g++'])
+        else:
+            deps.extend(['gcc-c++'])
+
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel("Fail to install Package: %s" % package)
+
+        tarball_path = os.path.join(os.path.abspath(''),
+                                    "rawread.py.data/rawread.tar")
+        tarball = self.get_data('rawread.tar')
+        archive.extract(tarball_path, self.teststmpdir)
+        self.source = os.path.join(self.teststmpdir,
+                                   os.path.basename(
+                                       tarball.split('.tar')[0]))
+        os.chdir(self.source)
+        build.make(self.source)
+
+        self.disk = self.params.get('disk', default=None)
+
+    def test(self):
+        """
+        Run 'rawread' with its arguments
+        """
+        err_val = []
+        for val in range(23):
+            cmd = "./rawread -t %s %s " % (val, self.disk)
+            if process.system(cmd, shell=True, ignore_status=True):
+                err_val.append(val)
+
+        if err_val:
+            self.fail("test failed for values : %s" % err_val)
```

Comment:
`" ".join(err_val)`

Reply:
why do we need this ?
err_val is a list

Reply:
You have the question and answer

Reply:
val is integer here, you cannot join with a string.
if you convert val to string and join then the output what we see on fail is no better than without using join.

following are the different scenarios
-------------------------------------------------
1) with current code : (without any join)

===>> FAIL: test failed for values : [0, 1]
this is very clean.

2) without converting val to string and just use join as u given
===>>> ERROR: sequence item 0: expected str instance, int found

3. converting the val to string while we append to err_val itself
======>>>> FAIL: test failed for values : 0 1

4. converting list itself to string while printing
======>>> FAIL: test failed for values : [ 0 ,   1 ]
this doesn't make big difference.

my opinion: we are making it more complex by adding the space before each element which is by default we have one space in list element.
let me know urs.

Reply:
@Naresh-ibm 

I never talked about converting `val` here. That is your assumption.
I am talking only about `err_val`.

What I want you to do is: when printing a list, format it into string and print, since you are using %s.